lambdas/journeyengine/src/main/java/uk/gov/di/ipv/core/journeyengine/JourneyEngineHandler.java

```diff
@@ -142,8 +148,13 @@ private JourneyEngineResult executeJourneyEvent(
                             new JourneyResponse(criStartUri + UK_PASSPORT_CRI_ID));
                     break;
                 case CRI_UK_PASSPORT:
-                    updateUserState(CRI_ADDRESS, ipvSessionItem);
-                    builder.setJourneyResponse(new JourneyResponse(criStartUri + ADDRESS_CRI_ID));
+                    if (journeyStep.equals(NEXT.toString())) {
+                        updateUserState(CRI_ADDRESS, ipvSessionItem);
+                        builder.setJourneyResponse(
+                                new JourneyResponse(criStartUri + ADDRESS_CRI_ID));
+                    } else if (journeyStep.equals(ERROR.toString())) {
+                        builder.setJourneyResponse(new JourneyResponse(journeyEndUri));
+                    }
```

Comment:
I'm wondering if we should throw here in an `else` block. I know we've validated that the step is valid, but if we add more steps to `VALID_JOURNEY_STEPS` we could have a situation where we fall through this `if` statement without actually doing anything.

Reply:
Yeah good shout, I'll update this

Reply:
I think dan is planning to change this as it needs to actually be checking for the /fail journey step. So this is probs fine to stay in an else if, as my PR will add a catch all else below this for the /error step

Reply:
Correct, just waiting on Danny's https://github.com/alphagov/di-ipv-core-back/pull/284 to be merged first.